test/MinimalDelegation.t.sol

```diff
@@ -17,23 +17,103 @@ contract MinimalDelegationTest is DelegationHandler {
         bytes32 keyHash = mockSecp256k1Key.hash();
 
         minimalDelegation.authorize(mockSecp256k1Key);
+        vm.snapshotGasLastCall("authorize");
 
         Key memory fetchedKey = minimalDelegation.getKey(keyHash);
         assertEq(fetchedKey.expiry, 0);
         assertEq(uint256(fetchedKey.keyType), uint256(KeyType.Secp256k1));
         assertEq(fetchedKey.isSuperAdmin, true);
         assertEq(fetchedKey.publicKey, abi.encodePacked(mockSecp256k1PublicKey));
+        assertEq(minimalDelegation.keyCount(), 1);
+    }
+
+    function test_authorize_expiryUpdated() public {
+        bytes32 keyHash = mockSecp256k1Key.hash();
+
+        minimalDelegation.authorize(mockSecp256k1Key);
+
+        Key memory fetchedKey = minimalDelegation.getKey(keyHash);
+        assertEq(fetchedKey.expiry, 0);
+        assertEq(uint256(fetchedKey.keyType), uint256(KeyType.Secp256k1));
+        assertEq(fetchedKey.isSuperAdmin, true);
+        assertEq(fetchedKey.publicKey, abi.encodePacked(mockSecp256k1PublicKey));
+        assertEq(minimalDelegation.keyCount(), 1);
+
+        mockSecp256k1Key =
```

Comment:
is this just changing the expiry? if so can do:

`mockSecp256k1Key.expiry = uint40(block.timestamp + 3600);`